src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/StaticHttpSettings.cs

```diff
@@ -0,0 +1,41 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Net.Http
+{
+    internal static class StaticHttpSettings
+    {
+        private const string AllowLatin1CharactersEnvironmentVariableSettingName = "DOTNET_SYSTEM_NET_HTTP_SOCKETSHTTPHANDLER_ALLOWLATIN1HEADERS";
+        private const string AllowLatin1CharactersAppCtxSettingName = "System.Net.Http.SocketsHttpHandler.AllowLatin1Headers";
+
+        private static readonly Lazy<bool> s_allowLatin1Headers = new Lazy<bool>(GetAllowLatin1HeadersSetting);
```

Comment:
I don't think this needs to use Lazy.  Any code that accesses StaticHttpSettings is going to trigger this, so it may as well just be:
```C#
private static readonly bool s_allowLatin1Headers = GetAllowLatin1HeadersSetting();
```

Reply:
I was afraid that type loading may trigger even earlier, preventing in-process calls of `AppContext.SetSwitch()` and `Environment.SetEnvironmentVariable()` to take effect.

Reply:
The tests seem to work fine after removing Lazy, so I pushed the change.